File: Backend/src/main/java/com/resumebuilder/backend/dto/AtsAnalysisResponse.java
package com.resumebuilder.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AtsAnalysisResponse {
    private int score;
    private String rating;
    private List<String> layoutSuggestions;
    private List<String> keywordSuggestions;
    private List<String> metricSuggestions;
    private List<String> verbSuggestions;
}

File: app/src/main/java/com/snehil/cvoptima/data/remote/model/AtsAnalysisResponse.kt
package com.snehil.cvoptima.data.remote.model

import com.google.gson.annotations.SerializedName

data class AtsAnalysisResponse(
    @SerializedName("score") val score: Int,
    @SerializedName("rating") val rating: String,
    @SerializedName("layoutSuggestions") val layoutSuggestions: List<String>?,
    @SerializedName("keywordSuggestions") val keywordSuggestions: List<String>?,
    @SerializedName("metricSuggestions") val metricSuggestions: List<String>?,
    @SerializedName("verbSuggestions") val verbSuggestions: List<String>?
)
